### PR TITLE
Allow orbit web JAX-RS APIs to return Tasks

### PR DESCRIPTION
**Pull request by JoeHegarty**
This is a very basic first pass to allow orbit web to return tasks in JAX-RS APIs.
Currently it blocks, this is required to correctly handle exceptions. In future I'd like to use code generation to avoid this, but I'd like to get the general programming model at least working so people can work against it.

### CODE REVIEW

File: web/pom.xml [hunk 1]
```diff
@@ -95,7 +95,7 @@
 
         <dependency>
             <groupId>org.glassfish.jersey.containers</groupId>
-            <artifactId>jersey-container-servlet-core</artifactId>
+            <artifactId>jersey-container-servlet</artifactId>
```
Comment (JoeHegarty) [inline web/pom.xml]: This makes the servlet 3 stuff available

File: web/src/main/java/com/ea/orbit/web/provider/TaskProvider.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+            Object result = entity;
+
+            try
+            {
+                result = ((Task<?>)entity).join();
+            }
+            catch(CompletionException e)
+            {
+                // We need to get the actual exception and rethrow it
+                throw (RuntimeException) e.getCause();
```
Comment (JoeHegarty) [inline web/src/main/java/com/ea/orbit/web/provider/TaskProvider.java]: This is so the actual thrown exception is the one which is passed to Jersey.

File: web/src/main/java/com/ea/orbit/web/provider/TaskProvider.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+            }
+
+            // Set generic type to null
+            // do we need to do anything special for wrapped types?
+            context.setGenericType(null);
```
Comment (JoeHegarty) [inline web/src/main/java/com/ea/orbit/web/provider/TaskProvider.java]: I may need to do some work here if you have a nested generic type like Task<SomeGeneric<String>> or similar, going to test this.

Comment (JoeHegarty) [inline web/src/main/java/com/ea/orbit/web/provider/TaskProvider.java]: Seems this actually works just fine, I've added a test covering it

File: web/src/main/java/com/ea/orbit/web/provider/TaskProvider.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+            {
+                result = ((Task<?>)entity).join();
+            }
+            catch(CompletionException e)
+            {
+                // We need to get the actual exception and rethrow it
+                throw (RuntimeException) ((e.getCause() != null) ? e.getCause() : e);
+            }
+
+            context.setGenericType(null);
```
Comment (DanielSperry) [inline web/src/main/java/com/ea/orbit/web/provider/TaskProvider.java]: We need to investigate the consequences of setGenericType(null)
I'd be more inclined to:
```
                        context.setGenericType(entityType == null || GenericTypeReflector.isSuperType(entityType, e.getClass())
                                // the actual entity class is narrower than task type parameter
                                // or the task type parameter is undefined or too generic
                                ? e.getClass()
                                // the entity type contains less information than the type parameter
                                // happens for instance if:
                                //    Task<List<SomeDto>> someMethod() { return Task.from(new ArrayList<>(value)) };
                                // where the entity class erases the type parameter
                                : entityType);
```

**Pull request merged**